### PR TITLE
CFE-2640: Run OS X unit tests in travis

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

File: travis-scripts/script.sh [hunk 1]
```diff
@@ -36,7 +40,7 @@ then
     make CFLAGS="-Werror -Wno-pointer-sign" -k
 elif [ "$JOB_TYPE" = compile_and_unit_test ]
 then
-    make CFLAGS="-Werror -Wno-pointer-sign"  &&
+    make CFLAGS="-Werror -Wno-pointer-sign"
```
Comment (vpodzime) [inline travis-scripts/script.sh]: Why this change?

**Comment by vpodzime** [comment]
> @vpodzime Do you know why we test for sparse file support?

No idea, doesn't look like a unit test at all to me.

File: travis-scripts/script.sh (hunk 1 above)
Comment (olehermanse) [reply travis-scripts/script.sh]: It seemed unneccessary, because of `set -e`?
